### DISCUSSION TITLE
No more rust contribution for the time being

### DISCUSSION THREAD

**Discussion by SamuelYvon** [SamuelYvon/netifaces-2#31] [Announcements]
Attention!

I will not be **merging** any PR for the Rust code base at the moment. The reason being is that I am splitting this repo in two, to contain the rust only code in an alternative repository. I will link the two repositories when I am done migrating.

Feel free to **open** issues **or** PRs, but note that I will **not** merge them, but they may be incorporated in the other repository, if it makes sense.

**Comment by relativityspace-jsmith** [comment]
@SamuelYvon I wanted to check in, what's the status of this effort? I am still really hoping to switch `multicast_expert` over to this library. However, there are still a few missing features causing issues; namely lack of netmask detection and default gateway detection on Windows. I could fix those issues right now if we were willing to bundle our own patched version of the get_adapters_addresses crate (as I have not been able to get in touch with that crate's developer to add netmask support). However, I don't know if that's acceptable to people, and I also don't know if that contribution would be accepted.

Anyway, I wanted to check in. Is moving the Rust code to another repo still happening? Is it something I could contribute to? If not, will development of this package continue as a rust + python package?

Thanks, Jamie

**Reply by SamuelYvon** [reply]
Hi Jamie!

Sorry for the delay. I was gone in the last two weeks. I still want to move this to a split repo (python interface, rust crate). I have the rust crate, but not the time to work on it. I can definitely give you access to it and we can chat about how to go about it.

**Reply by SamuelYvon** [reply]
I just checked and you already have access. There's a public email on my profile. Feel free to reach out.

**Reply by relativityspace-jsmith** [reply]
Ah I see it now!

However, there is a bit of a wrinkle. My colleague at work actually just today mentioned the `netdev` crate, which seems like it actually does pretty much the same thing as your in-progress one. I looked through the code and feature set a bit and it... looks good, actually.

In the interest of reaching my goal faster (replacement for netifaces in Python), I have to ask... is there a reason not to simply create a Python library that wraps this crate instead of implementing all of our own logic in Rust?

**Reply by SamuelYvon** [reply]
I have no attachment to my current rust crate. If you can provide bindings to `netdev` I'm happy to let go of my rust version and bind to `netdev`
